### ISSUE TITLE
SilverBullet will download all the markdown files when first open. That would be slow.

### ISSUE THREAD

**Issue by wishWinds** [silverbulletmd/silverbullet#461]
I have many md files. I found when I first open the website, it's much slower and command+k not work at the beginning, and I found browser will download all the md files even I not request the files.

I think it will be great if silverbullet only transfer the files if user request.

**Comment by zefhemel** [comment]
This is how the earlier versions of SB worked: all files and indexes were kept on the server, the client only fetched the files that were specifically requested and no indexes were kept in the client. This worked, but did not support the offline use case. If the server went away, or there's no network connection to the server, you wouldn't have access to your notes.

This lead to the current architecture where a full copy of all content is synced to the client (one time operation) and all indexing and processing happens there. The cost of this is what you describe: all content needs to be synced not only for offline access, but also to give features like directives, page renaming complete information to do their job. 

For people with spaces that aren't super large, this is not a problem. I have around 800 .md files myself, and it takes a while to sync on a new client (but again: this is only on first load), but it's not too bad. If you have thousands of files, this would indeed become problematic. How many files do you have?

**Comment by wishWinds** [comment]
Hi

Thanks for your explain. Totally understand.

I have around 300 md files, but the machine is about 8GB memory. and sometimes I will clean all chrome caches. I think this will make SB sync all md files again.

I found a way to work around. I use syncthing to sync the files, and use local SB client to open my Notes Directory. 

BTW, This is a nice project. I used for my personal knowledge management. Thanks for your great job.
